error while running `make build-local`

**What happened**:

while running `make build-local` i met follow error
```
 network-problem-detector git:(main) ✗ make build-local
# github.com/gardener/network-problem-detector/cmd/nwpd
runtime.libcCall: nosplit stack overflow
	792	assumed on entry to syscall.ptrace<1> (nosplit)
	680	after syscall.ptrace<1> (nosplit) uses 112
	672	on entry to syscall.ptrace1<1> (nosplit)
	480	after syscall.ptrace1<1> (nosplit) uses 192
	472	on entry to syscall.syscall6<1> (nosplit)
	288	after syscall.syscall6<1> (nosplit) uses 184
	280	on entry to runtime.entersyscall<1> (nosplit)
	256	after runtime.entersyscall<1> (nosplit) uses 24
	248	on entry to runtime.reentersyscall<1> (nosplit)
	192	after runtime.reentersyscall<1> (nosplit) uses 56
	184	on entry to runtime.casgstatus<1> (nosplit)
	120	after runtime.casgstatus<1> (nosplit) uses 64
	112	on entry to runtime.nanotime<1> (nosplit)
	96	after runtime.nanotime<1> (nosplit) uses 16
	88	on entry to runtime.nanotime1<0> (nosplit)
	48	after runtime.nanotime1<0> (nosplit) uses 40
	40	on entry to runtime.libcCall<1> (nosplit)
	-16	after runtime.libcCall<1> (nosplit) uses 56
make: *** [build-local] Error 2
```

i googled the error and ref https://github.com/golang/go/issues/51247 , then i removed `-race \` from MakeFile target `build-local` and then i can build it.

My env:

```
 network-problem-detector git:(main) ✗ go version
go version go1.18.5 darwin/amd64
➜  network-problem-detector git:(main) ✗ system_profiler SPSoftwareDataType
Software:

    System Software Overview:

      System Version: macOS 12.4 (21F79)
      Kernel Version: Darwin 21.5.0
      Boot Volume: Macintosh HD
      Boot Mode: Normal
      Computer Name: xxx
      User Name: xxx
      Secure Virtual Memory: Enabled
      System Integrity Protection: Enabled
      Time since boot: 21 days 17:30
```

**What you expected to happen**:

`make build-local` should success

**How to reproduce it (as minimally and precisely as possible)**:

run `make build-local`

**Anything else we need to know**:

**Environment**:
see above

--- a/.ci/pipeline_definitions
+++ b/.ci/pipeline_definitions
@@ -21,7 +21,7 @@ network-problem-detector:
             dockerfile: 'Dockerfile'
     steps:
       verify:
-        image: golang:1.18.2
+        image: golang:1.18.5
   jobs:
     head-update:
       traits:

--- a/Dockerfile
+++ b/Dockerfile
@@ -3,7 +3,7 @@
 # SPDX-License-Identifier: Apache-2.0
 
 ############# builder
-FROM golang:1.18.3 AS builder
+FROM golang:1.18.5 AS builder
 
 WORKDIR /build
 COPY . .

--- a/Makefile
+++ b/Makefile
@@ -38,7 +38,6 @@ build-local:
 	@CGO_ENABLED=1 GO111MODULE=on go build -o $(EXECUTABLE) \
 	    -race \
         -mod=vendor \
-	    -gcflags="all=-N -l" \
 	    -ldflags "-X 'main.Version=$(EFFECTIVE_VERSION)' -X 'main.ImageTag=$(IMAGE_TAG)'"\
 	    ./cmd/nwpd
 

